### ISSUE TITLE
Audio sync exists with 4.9.0 (how to reproduce)

### ISSUE THREAD

**Issue by loopy750** [DistroAV/DistroAV#498]
Audio sync is still an issue with 4.9.0, and tested using a few different network cards. The issue can be easily reproduced with both Normal and Low latency modes, although it may be less severe with the latter.

Please see the videos for how to reproduce, and the files used (linked/attached).

Latency: Normal - https://streamable.com/5p56f8
Latency: Low - https://streamable.com/b0dbsc

Over time, the audio/video may slowly drift out of sync. However, we can cause this effect immediately using the "static" image.

https://i.ibb.co/KGbPyPP/Image4.jpg - This is the expected result: frame "1" matches beginning of the 1kHz tone, hence in sync.

https://i.ibb.co/2Nch2Z6/Image3.jpg - The result after the "static" image has been shown: 1kHz tone begins at frame "54", 7 frames out of sync, or 116ms.

"Stress test" image file: https://i.ibb.co/SQhKMKQ/400pc-Gaussian-50pc-Grey.png
60 FPS Test mp4: [60 FPS TEST v2_1 START_AT_1.mp4_13MB.zip](https://github.com/Palakis/obs-ndi/files/4650132/60.FPS.TEST.v2_1.START_AT_1.mp4_13MB.zip)

**Comment by WiredUK** [comment]
I'd love to know if this is an NDI thing or an OBS thing. We're also seeing audio sync issues, though the low-latency mode appears to be a lot better. Even if I had a way to "reset" the sync, that would be enough for us, it's really starting to become a big problem.

**Comment by Anonymous2** [comment]
Hello, I also found a lot of audio/video desync with 4.9.0 that I never had on previous versions

**Comment by manuelnyc** [comment]
@WiredUK the problem is THIS palakis NDI plugin, when I use newtek tools I never have an issue, no drops no nothing, but when I use this palakis the problems, frame drops, out of sync, start to happen, sadly this is the only NDI plugin for OBS

**Comment by WiredUK** [comment]
> @WiredUK the problem is THIS palakis NDI plugin, when I use newtek tools I never have an issue, no drops no nothing, but when I use this palakis the problems, frame drops, out of sync, start to happen, sadly this is the only NDI plugin for OBS

Not necessarily. It could still be OBS. We really need to understand this, it's really hurting the quality of our streams.

**Comment by manuelnyc** [comment]
> > @WiredUK the problem is THIS palakis NDI plugin, when I use newtek tools I never have an issue, no drops no nothing, but when I use this palakis the problems, frame drops, out of sync, start to happen, sadly this is the only NDI plugin for OBS
> 
> Not necessarily. It could still be OBS. We really need to understand this, it's really hurting the quality of our streams.


the only solution I found is this: (it's not a solution, it's a trick)

I produce the whole thing in OBS, skype calls etc, then from this computer I enable OBS's NDI output to my streaming computer, my streaming computer that receives my NDI stream is running Wirecast....... so to avoid the video/audio sync problems, every time there's a 1 or 2 seconds silence in the stream, I click reconnect in wirecast so the stream is fresh again...... doing this I never had a sync issue anymore, tested yesterday on a 2 hrs long stream 

it is what it is,.......either this plugin or the whole NDI tech thing is still not well developed yet

**Comment by dgatwood** [comment]
One problem I just saw involved feeding 29.97/59.94 fps output from a Tricaster into OBS with OBS configured for 30 fps video.  The video got farther and farther behind.  What was interesting is that feeding 59.94 fps video into a 30 fps configuration drifted about the same amount as 29.97 fps video did.

So this seems to be a problem with some part of the NDI to OBS data path not handling the difference between drop-frame and non-drop-frame modes, but still getting the 30 fps vs 60 fps portion of the frame rate right.

Switching OBS to 29.97 fixed the problem.

**Comment by dgatwood** [comment]
It's worth noting that if things are drifting out of sync less with low latency mode, that usually points to packet loss caused by a bad cable.  In testing network cables from some seemingly reputable manufacturers, I've seen more than one out of every three cables fail packet loss tests at gigabit speeds, which means there's a chance that they will fall back to 100-megabit speeds and ruin your production.  Every time I've had sync problems, I eventually traced it back to a bad cable.

In theory, you **can** configure the plug-in to use the camera's time stamp, but that requires all of the cameras have the correct time, which on an unconnected network isn't likely to be the case unless you also run an NTP server on that network, so this mode is likely to cause synchronization problems that appear immediately and never go away no matter what you do.

In the other synchronization mode, where time stamps are generated by when packets are received, you have a different set of synchronization problems.  Everything is perfect until you lose packets, and then things go wrong.  Because the time stamps are based on when the packet is received, not when it is sent, and because the cameras are using TCP instead of UDP, when a packet gets lost, the packet gets resent, so now you have video that was sent a fraction of a second ago, but appears to have been sent "now", which means that camera is now behind all of the other cameras.  The more times this happens, the farther behind the camera gets.

It may be possible to avoid this problem if all of your cameras are based on NDI 5 and are configured to genlock from a single master source, though even then, if your master source desyncs, then all of your cameras will desync unless your audio is also coming from a genlocked NDI source.

To make a long story short, I've concluded that owning and periodically using a Fluke cable tester to test every cable in your collection is an absolute necessity with NDI, because it does *not* tolerate even a single packet getting lost.  If your cables don't test out good at 10-gigabit, throw them away, because there's still a chance they'll fail at gigabit speeds.

I've also started labeling my cables so that I have a known association between cameras and cables.  If a camera drifts out of sync, I can yank the numbered cable, wait a minute or so, and reconnect it, which will reset the connection and get that camera back in sync.  Then the cable goes in the garbage can, because that's pretty much always the problem.
